src/Module/AddBookingToCartHandler.php

```diff
@@ -151,15 +186,58 @@ protected function _addToCart(BookingInterface $booking)
         // Get the cart item data keys from the cart item config
         $eddBkKey     = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
         $bookingIdKey = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+        $priceIdKey   = $this->_containerGetPath($this->cartItemConfig, ['data', 'price_id_key']);
+
+        // Find the price ID
+        $service  = $this->_getServiceById($serviceId);
+        $lengths  = $this->_containerGet($service, 'session_lengths');
+        $duration = $booking->getDuration();
+        $priceId  = null;
+        foreach ($lengths as $_idx => $_lengthInfo) {
+            $_length = (int) $this->_containerGet($_lengthInfo, 'sessionLength');
+
+            if ($_length === $duration) {
+                $priceId = $_idx;
+                break;
+            }
+        }
 
         // Create the cart item data
         $data = [
-            $eddBkKey => [
+            $priceIdKey => $priceId,
+            $eddBkKey   => [
                 $bookingIdKey => $bookingId,
             ],
         ];
 
         // Add the service (EDD Download) to the cart with the additional EDD Bookings data
         $this->eddCart->add($serviceId, $data);
     }
+
+    /**
+     * Retrieves a booking by ID.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable $serviceId The ID of the service to retrieve.
+     *
+     * @return array|stdClass|ArrayAccess|ContainerInterface|null The service instance, or null if no service was
+     *                                                            found for the given ID.
+     */
+    protected function _getServiceById($serviceId)
+    {
+        $b = $this->exprBuilder;
+
+        $services = $this->servicesSelectRm->select(
+            $b->and(
+                $b->eq($b->ef('service', 'id'), $b->lit($serviceId))
```

Comment:
On one hand, the name of the entity is nicely encapsulated in the RM. On the other, consuming code kinda seems to need to know that name in order to select by its fields.

Reply:
Well, it's also possible replace `$b->ef('service', 'id)` with `$b->var('id')`. But even still, knowing the name is already a decent abstraction from knowing the table (or even storage medium at that!).

Reply:
I was thinking that maybe we should add the ability to retrieve the entity name from the RM in the future.